ecdsa/keygen/round_1.go

```diff
@@ -66,7 +67,10 @@ func (round *round1) Start() *tss.Error {
 	// 9-11. compute ntilde, h1, h2 (uses safe primes)
 	// use the pre-params if they were provided to the LocalParty constructor
 	var preParams *LocalPreParams
-	if round.save.LocalPreParams.Validate() {
+	if round.save.LocalPreParams.Validate() && !round.save.LocalPreParams.ValidateWithProof() {
+		return round.WrapError(
+			errors.New("`optionalPreParams` failed to validate; it might have been generated with an older version of tss-lib"))
+	} else if round.save.LocalPreParams.ValidateWithProof() {
```

Comment:
this can be `round.save.LocalPreParams.Validate() && !round.save.LocalPreParams.ValidateWithProof()`

Reply:
The first condition on line 70 checks whether they are trying to use an old pre-params from tss-lib before this PR. The second condition on line 73 will pass if they are using pre-params generated with the new tss-lib post-PR.